src/main/index.ts

```diff
@@ -146,9 +158,19 @@ if (!isFirstInstance) {
   };
 
   app.on('session-created', (session) => {
+    let reservedStateValues = [];
     if (!isDevelopment) {
       log.info('Cleaning local storage on session creation');
-      session.clearStorageData({ storages: ['localstorage'] }).catch(log.error);
+
+      if (mainWindow != null) {
+        mainWindow.webContents.executeJavaScript('({...localStorage});', true).then((localStorage) => {
+          reservedStateValues = reservedStates.map((state) => localStorage.getItem(state));
+          session.clearStorageData({ storages: ['localstorage'] }).catch(log.error);
```

Comment:
This is a Promise, meaning it's asynchronous, so you shouldn't be setting items just yet.